### DISCUSSION TITLE
Why is homebrew weechat trying to delete a file from brew's local copy of git?

### DISCUSSION THREAD

**Discussion by robolange** [Homebrew/discussions#5479] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.8
ORIGIN: https://github.com/Homebrew/brew
HEAD: e5f776b3e23cae8f1b6c3d1194b8329bf701965f
Last commit: 8 days ago
Core tap JSON: 06 Jul 01:22 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :1
HOMEBREW_MAKE_JOBS: 20
Homebrew Ruby: 3.3.3 => /home/linuxbrew/.linuxbrew/Cellar/ruby/3.3.3/bin/ruby
CPU: 20-core 64-bit unknown_0x6_0xba
Clang: N/A
Git: 2.39.2 => /bin/git
Curl: 7.88.1 => /bin/curl
Kernel: Linux 6.7.12+bpo-amd64 x86_64 GNU/Linux
OS: Debian GNU/Linux 12 (bookworm) (bookworm)
Host glibc: 2.36
/usr/bin/gcc: 12.2.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I have installed Linux homebrew with one user, and I run the brew apps using another user, my regular user. This has worked for years and I haven't had any issues. I use homebrew on Linux only for certain specific frequently updated apps, like weechat, but mostly use only distribution-provided apps.

Recently, within the past few months, I've noticed that homebrew's weechat has been behaving oddly. Whenever I start it, I get a warning written to the terminal that it tried and failed to delete a file:

```
 % weechat
rm: cannot remove '/home/linuxbrew/.linuxbrew/Homebrew/.git/describe-cache/e5f776b3e23cae8f1b6c3d1194b8329bf701965f': Permission denied
```

Weechat operates normally, with the exception of this warning line. The permission denied happens because my regular user does not have permission to alter or delete homebrew files. But, why is weechat trying to delete a file in homebrew's git local copy? Why would it even know/care that that directory structure exists?

(For the record, the file contains only the line `4.3.8`, which looks like a version number, but it's not the current version of weechat, which is currently 4.3.4.)

If I manually remove the file and its parent directory using my privileged user, then the warning line stops appearing when I run weechat, and only appears again after the next time I update homebrew. But once again, why is homebrew weechat aware of this directory structure, and why is it trying to delete a file from it?

**Comment by gromgit** [comment, marked as answer]
AFAIK, there's nothing about _normal_ `weechat` that would operate on Homebrew itself, and the symptoms you describe sound more like you're running `brew <something_or_other>` in your prompt command. If you are, you probably want to find another way to get what you need.

> (For the record, the file contains only the line `4.3.8`, which looks like a version number, but it's not the current version of weechat, which is currently 4.3.4.)

It's the version of Homebrew you're running (see your `brew config` output), which is expected given the pathname you mentioned.

**Reply by robolange** [reply]
Thanks. You reminded me that years ago I had aliased weechat to a script that ran a brew command:
```
brew_deps=$(brew deps --direct --quiet weechat)
```
This command made a list of dependencies, which I used to set the paths to scripting languages appropriately, as I do not have the general brew bin directory at the front of my path. Then it called the real weechat.

I guess in the past few releases of brew, that brew command wants to remove the cache file. Makes sense, thanks.

**Comment by MikeMcQuaid** [comment]
You can ignore this warning but it'll be hidden when https://github.com/Homebrew/brew/pull/18084 is merged.
